src/arrayMethodSort.js

```diff
@@ -4,8 +4,42 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const length = this.length;
+
+    let comparator;
+
+    if (typeof compareFunction !== 'function') {
+      comparator = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA < strB) {
+          return -1;
+        }
+
+        if (strA > strB) {
+          return 1;
+        }
+
+        return 0;
+      };
+    } else {
+      comparator = compareFunction;
+    }
+
+    for (let i = 0; i < length - 1; i++) {
+      for (let j = 0; j < length - 1 - i; j++) {
+        if (comparator(this[j], this[j + 1]) > 0) {
+          const temp = this[j];
+
+          this[j] = this[j + 1];
+          this[j + 1] = temp;
+        }
+      }
+    }
+
+    return this;
   };
```

Comment:
Your sort algorithm is a Bubble Sort which has a worst-case and average complexity of O(n^2). For small datasets, it's okay, but for larger datasets, you might want to consider a more efficient algorithm like Quick Sort or Merge Sort. However, since the task does not specify the sorting algorithm to be used, this is just a recommendation.